dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
+        if gather_statistics is None:
+            gather_statistics = True
+        if split_row_groups is None:
+            split_row_groups = True
+
+    # Step 2: Construct necessary (parquet) partitioning information
+    partition_info = {"partitions": None, "partition_keys": {}, "partition_names": []}
+    fn_partitioned = False
+    if dataset.partitions is not None:
+        fn_partitioned = True
+        partition_info["partition_names"] = [
+            n for n in dataset.partitions.partition_names if n is not None
+        ]
+        partition_info["partitions"] = dataset.partitions
+        for piece in dataset.pieces:
+            partition_info["partition_keys"][piece.path] = piece.partition_keys
+
+    # Step 3: Construct a single `metadata` object. We can
+    #         directly use dataset.metadata if it is available.
+    #         Otherwise, if `gather_statistics` or `split_row_groups`,
+    #         we need to gether the footer metadata manually
+    metadata = None
+    if dataset.metadata:
+        # We have a _metadata file.
+        # PyArrow already did the work for us
+        schema = dataset.metadata.schema.to_arrow_schema()
+        if gather_statistics is None:
+            gather_statistics = True
+        if split_row_groups is None:
+            split_row_groups = True
+        return (
+            schema,
+            dataset.metadata,
+            base,
+            partition_info,
+            split_row_groups,
+            gather_statistics,
         )
-        # Check if the column-chunk file_path's are set in "_metadata".
-        # If available, we can use the path to sort the row-groups
-        col_chunk_paths = False
-        if dataset.metadata:
-            col_chunk_paths = all(
-                dataset.metadata.row_group(i).column(0).file_path is not None
-                for i in range(dataset.metadata.num_row_groups)
+    else:
+        # No _metadata file.
+        # May need to collect footer metadata manually
+        if dataset.schema is not None:
+            schema = dataset.schema.to_arrow_schema()
+        else:
+            schema = None
+        if gather_statistics is None:
+            gather_statistics = False
+        if split_row_groups is None:
+            split_row_groups = False
+        metadata = None
+        if not (split_row_groups or gather_statistics):
+            # Don't need to construct real metadata if
+            # we are not gathering statistics or splitting
+            # by row-group
+            metadata = [p.path for p in dataset.pieces]
+            if schema is None:
+                schema = dataset.pieces[0].get_metadata().schema.to_arrow_schema()
+            return (
+                schema,
+                metadata,
+                base,
+                partition_info,
+                split_row_groups,
+                gather_statistics,
             )
+        for piece, fn in zip(dataset.pieces, fns):
```

Comment:
comment: this is the only path where we might (must?) scan all the files - no _metadata, and the user wants either splitting or statistics explicitly